server/api/rounds.py

```diff
@@ -399,34 +399,64 @@ def is_round_complete(election: Election, round: Round) -> bool:
-                and_(
-                    JurisdictionResult.jurisdiction_id == Jurisdiction.id,
-                    JurisdictionResult.contest_id == Contest.id,
-                    JurisdictionResult.round_id == round.id,
-                ),
+        num_jurisdictions_without_results: int
+        # Special case: if we sampled all ballots, we just check every
+        # jurisdiction in the targeted contest's universe
+        if sampled_all_ballots(round, election):
+            num_jurisdictions_without_results = (
+                Contest.query.filter_by(election_id=election.id, is_targeted=True)
+                .join(Contest.jurisdictions)
+                .outerjoin(
+                    JurisdictionResult,
+                    and_(
+                        JurisdictionResult.jurisdiction_id == Jurisdiction.id,
+                        JurisdictionResult.contest_id == Contest.id,
+                        JurisdictionResult.round_id == round.id,
+                    ),
+                )
+                .group_by(Jurisdiction.id, Contest.id)
+                .having(func.count(JurisdictionResult.result) == 0)
+                .count()
+            )
+        else:
+            num_jurisdictions_without_results = (
+                # For each jurisdiction...
+                Jurisdiction.query.filter_by(election_id=election.id)
+                # Where ballots were sampled...
+                .join(Jurisdiction.batches)
+                .join(Batch.ballots)
+                # And those ballots were sampled this round...
+                .join(SampledBallot.draws)
+                .filter_by(round_id=round.id)
+                .join(SampledBallotDraw.contest)
+                # Count the number of results recorded for each contest.
+                .outerjoin(
+                    JurisdictionResult,
+                    and_(
+                        JurisdictionResult.jurisdiction_id == Jurisdiction.id,
+                        JurisdictionResult.contest_id == Contest.id,
+                        JurisdictionResult.round_id == round.id,
+                    ),
+                )
+                # Finally, count the number of jurisdiction/contest pairs for which
+                # there is no result recorded
+                .group_by(Jurisdiction.id, Contest.id)
+                .having(func.count(JurisdictionResult.result) == 0)
+                .count()
             )
-            # Finally, count the number of jurisdiction/contest pairs for which
-            # there is no result recorded
-            .group_by(Jurisdiction.id, Contest.id)
-            .having(func.count(JurisdictionResult.result) == 0)
-            .count()
-        )
         return num_jurisdictions_without_results == 0
 
 
+def sampled_all_ballots(round: Round, election: Election):
+    return election.audit_type == AuditType.BALLOT_POLLING and any(
+        round_contest.sample_size >= round_contest.contest.total_ballots_cast
```

Comment:
This code seems to set the threshold for "more efficient to audit all the ballots" at a sample size equal to `total_ballots_cast`. I think it is more likely to be a tenth of that or less, but the threshold varies by jurisdiction and circumstances. You might want to multiply by a fraction here which can be configured, e.g. via an environmental variable.

Reply:
Agreed with the goal, but this is not where that happens. The decision to offer "all in" happens in `audit_math/bravo.py` based on a 25% threshold (hard-wired for now, we can configure it later), at which point the sample size selected is all the ballots, and then this code here triggers.